src/components/custom/ItemRollSimulator.tsx

```
@@ -51,58 +69,185 @@ const ItemRollSimulator: React.FC<ItemRollSimulatorProps> = ({ item, trigger })
-
-                overall += percentage;
-                identificationCount++;
-
-                result[stat] = {
-                    raw: finalRoll,
-                    percentage: Number(percentage.toFixed(1)),
-                    star,
-                };
+        for (const [idName, idValue] of Object.entries(IDs)) {
+            if (typeof idValue === 'object' && 'raw' in idValue) {
+                if (selectedAugment === "Corkian Insulator" && lockedIdentification === idName && RolledIdentifications[idName] && RolledIdentifications[idName].hasOwnProperty('percentage')) {
```

Biome (1.9.4)

[error] Do not access Object.prototype method 'hasOwnProperty' from target object.

It's recommended using Object.hasOwn() instead of using Object.hasOwnProperty().
See MDN web docs for more details.

(lint/suspicious/noPrototypeBuiltins)